test/JsonApiDotNetCoreMongoDbExampleTests/IntegrationTests/ReadWrite/WorkItem.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Collections.Generic;
+using JsonApiDotNetCore.MongoDb.Resources;
+using JsonApiDotNetCore.Resources.Annotations;
+using MongoDB.Bson.Serialization.Attributes;
+
+namespace JsonApiDotNetCoreMongoDbExampleTests.IntegrationTests.ReadWrite
+{
+    public sealed class WorkItem : MongoDbIdentifiable
+    {
+        [Attr]
+        public string Description { get; set; }
+
+        [Attr]
+        public DateTimeOffset? DueAt { get; set; }
+
+        [Attr]
+        public WorkItemPriority Priority { get; set; }
+
+        [BsonIgnore]
+        [Attr(Capabilities = ~(AttrCapabilities.AllowCreate | AttrCapabilities.AllowChange))]
+        public Guid ConcurrencyToken
+        {
+            get => Guid.NewGuid();
+            set { }
+        }
+
+        [HasOne]
+        public UserAccount Assignee { get; set; }
+
+        [HasMany]
+        public ISet<UserAccount> Subscribers { get; set; }
+
+        [BsonIgnore]
+        [HasManyThrough(nameof(WorkItemTags))]
+        public ISet<WorkTag> Tags { get; set; }
+        public ICollection<WorkItemTag> WorkItemTags { get; set; }
+
+        [HasOne]
+        public WorkItem Parent { get; set; }
+
+        [HasMany]
+        public IList<WorkItem> Children { get; set; }
+
+        [BsonIgnore]
+        [HasManyThrough(nameof(RelatedFromItems), LeftPropertyName = nameof(WorkItemToWorkItem.ToItem), RightPropertyName = nameof(WorkItemToWorkItem.FromItem))]
+        public IList<WorkItem> RelatedFrom { get; set; }
+        public IList<WorkItemToWorkItem> RelatedFromItems { get; set; }
+
+        [BsonIgnore]
```

Comment:
In this file, it looks as if you replaced all occurrences of `[NotMapped]` with `[BsonIgnore]`. I wonder what that means for the other relationships (for example: Parent), that don't have BsonIgnore on them. Shoudl they? I guess that now the related resources will be stored inside the same document, but I don't know enough of MongoDB to reason about that.

I'm not saying changes are needed, just wanted to make you aware of this. At other places where relationships are used, you did add BsonIgnore on them, so I don't see a consistent approach in how to deal with that.

Reply:
The only relationships in which not having `BsonIgnore` is problematic are those which are recursive, regular relationships should be stored as nested docs as you say. In order to avoid any confusion I'd thought of just putting `BsonIgnore` in all of them since they're just being used to test that we get the proper error when trying to reference relationships so here is just an example of me forgetting to do just that.

Reply:
Thanks for the clarification.